src/algorithm/user_select/user_select.rs

```diff
@@ -0,0 +1,196 @@
+use std::str::FromStr;
+
+use crate::{algorithm::storage::storage::Storage, kernel::{dbgid::dbgid::DbgId, entities::{crane_work_area::CraneWorkArea, driver_type::DriverType, hoist_rope_balance_degree::HoistRopeBalanceDegree, hoist_rope_twisting_method::HoistRopeTwistingMethod, hoist_rope_type::HoistRopeType, liftclass::LiftClass, load_combination::LoadCombination, mechanism_work_type::MechanismWorkType, value::Value}}};
+
+///
+/// Класс, которой реализует хранение характеристики выбранные пользователем для дальнейшего расчета крана
+pub struct UserSelect {
+    pub dbgid: DbgId,
+    pub m_to_lift: f64,
+    pub mechanism_work_type: MechanismWorkType,
+    pub vhmax: f64,
+    pub vhcs: f64,
+    pub lift_class: LiftClass,
+    pub load_comb: LoadCombination,
+    pub drive_type: DriverType,
+    pub hook_type: String,
+    pub name_cargo_hand_device: String,
+    pub weight_cargo_hand_device: f64,
+    pub rejecting_blocks: f64,
+    pub crane_work_area: CraneWorkArea,
+    pub hoist_rope_core_type: HoistRopeType,
+    pub hoist_rope_twisting_method: HoistRopeTwistingMethod,
+    pub hoist_rope_balance_degree: HoistRopeBalanceDegree,
+    pub hoist_rope_diametr: f64,
+
+}
+//
+//
+//
+impl UserSelect {
+    ///
+    /// Метод создания экземпляра класса UserSelect
+    /// - storage - экземпляр класса-хранилища Storage, в котором находится "таблица" конструкций, кранов, подшипников
+    pub fn new(user_select_storage: &Storage) -> Self {
+        let mut m_to_lift_tmp = 0.0;
+        if let Some(value) = user_select_storage.get("конструкции/крюк/грузоподъемность/") {
+            if let Value::Data(data) = value {
+                m_to_lift_tmp = *data;
+            }
+        }
+
```

Comment:
try to avoid empty lines in the code blocks
if you feels empty line required, then possible you have different functionality to be moved to separate or another class